pkgs/applications/blockchains/monero-cli/default.nix

```diff
@@ -35,6 +35,28 @@ stdenv.mkDerivation rec {
   };
 
   patches = [
+    # cmake: remove unused/extera cmake/FindMiniupnpc.cmake and only rely on external/miniupnpc
+    # https://github.com/monero-project/monero/pull/9366
+    (fetchpatch2 {
+      url = "https://github.com/monero-project/monero/commit/5074a543a49f7e23fb39b6462fd4c4c9741c3693.patch?full_index=1";
+      hash = "sha256-dS2hhEU6m2of0ULlsf+/tZMHUmq3vGGXJPGHvtnpQnY=";
+    })
+
+    # cmake: add different parameters to add_monero_library.
+    # https://github.com/monero-project/monero/pull/9367
+    (fetchpatch2 {
+      url = "https://github.com/monero-project/monero/commit/b91ead90254ac6d6daf908f689c38e372a44c615.patch?full_index=1";
+      hash = "sha256-DL2YqkvEONbeEDqLOAo2eSF5JF5gOzKcLKeNlUXBY1w=";
+    })
+
+    # external: update miniupnpc to 2.2.8
+    # https://github.com/monero-project/monero/pull/9367
+    (fetchpatch2 {
+      url = "https://github.com/monero-project/monero/commit/d81da086ec5088a04b3f7b34831e72910300e2f7.patch?full_index=1";
+      hash = "sha256-ZJGiDMk5DMmEXwzoUYPC+DIoebluFh54kMQtQU78ckI=";
+      excludes = [ "external/miniupnp" ];
+    })
+
     ./use-system-libraries.patch
```

Comment:
A bit weird to me how these external patches don't allow us to not have also a local patch... However this shouldn't block this PR.

Reply:
The previous anti‐vendoring patch was actually not fully functional in terms of specifying the header locations. The upstream PRs that fix miniupnpc 2.2.8 also forced the vendoring even harder. So since it’ll need dealing with anyway on the next version, I just fixed the anti‐vendoring stuff and reduced its size in the process.